c10/core/TensorImpl.h

```diff
@@ -856,6 +855,7 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
 
     refresh_numel();
     refresh_contiguous();
+    reset_memory_format_tag();
```

Comment:
So, I'm guessing there are some of functions which semantically should preserve memory format tag, and they will have to fix things up after calling `set_sizes_and_strides`. For example, consider `torch.select` (e.g., indexing) on the batch dimension. It seems bad to lose knowledge that it was NHWC in this case.